opentelemetry-sdk/tests/metrics/test_periodic_exporting_metric_reader.py

```diff
@@ -0,0 +1,135 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import random
+import time
+from functools import partial
+from unittest.mock import Mock
+
+from opentelemetry.sdk._metrics.export import (
+    MetricExporter,
+    PeriodicExportingMetricReader,
+)
+from opentelemetry.sdk._metrics.point import Gauge, Metric, Sum
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.test.concurrency_test import ConcurrencyTestBase
+from opentelemetry.util._time import _time_ns
+
+
+class FakeProvider:
+    def __init__(self, readers):
+        self._metric_readers = readers
+
+    def register_periodic_reader(self, periodic_reader):
+        periodic_reader._register_meter_provider(self)
+
+
+class FakeMetricsExporter(MetricExporter):
+    def __init__(self, wait=0):
+        self.wait = wait
+        self.metrics = []
+
+    def export(self, metrics):
+        time.sleep(self.wait)
+        self.metrics += metrics
+        return random.choice([True, False])
```

Comment:
Make sure to create a seed and log it to be able to reproduce test results. :+1:

Reply:
This isn't really used anywhere. Updated